plausible-adapter  to send events in php

Pretty-19 opened the pull request:
- [x]  Forked & cloned this repository
- [x]   Added an event function according to the Event API  given https://plausible.io/docs/events-api#post-apievent

TorstenDittmann commented on src/Analytics/Plausible.php:
```diff
@@ -0,0 +1,55 @@
+  private const URL='https://plausible.io/api/event';
+
+
+  public function __construct(string $clientIP){
+    $clientIP=$_SERVER['REMOTE_ADDR'];
```
`$_SERVER['REMOTE_ADDR']` will only work in Server mode - but not in CLI.

For CLI mode we can fallback to `127.0.0.1`.
```suggestion
  public function __construct(string $clientIP = '127.0.0.1'){
```

TorstenDittmann commented on src/Analytics/Plausible.php:
```diff
@@ -0,0 +1,55 @@
+
+  public function __construct(string $clientIP){
+    $clientIP=$_SERVER['REMOTE_ADDR'];
+    $this->headers=array(' X_FORWARDED_FOR: '  .$clientIP , ' Content-Type: application/json ');
```
For better readability 👍🏻 

```suggestion
    $this->headers = [
        'X_FORWARDED_FOR: '  .$clientIP, 
        'Content-Type: application/json'
    ];
```

TorstenDittmann commented on src/Analytics/Plausible.php:
```diff
@@ -0,0 +1,55 @@
+         return false;
+     }
+    $query=[
+        '__name' =>'pageview',
+        '__url' => $url,
+        '__domain' =>$domain
```
Looking at the [API docs](https://plausible.io/docs/events-api) the underscores are not needed.

TorstenDittmann commented on src/Analytics/Plausible.php:
```diff
@@ -0,0 +1,55 @@
+       curl_setopt($ch,CURLOPT_URL,self::URL);
+       curl_setopt($ch,CURLOPT_POST,true);
+       curl_setopt($ch,CURLOPT_HTTPHEADER ,$this->headers);
+       curl_setopt($ch,CURLOPT_USERAGENT, $_SERVER['HTTP_USER_AGENT']);
```
`$_SERVER['HTTP_USER_AGENT']` will not work in CLI mode. You can make this part of the constructor of the class 👍🏻

Pretty-19 replied:
All the changes  are noted , will get to  fixing them  @TorstenDittmann  Thanks.

TorstenDittmann reviewed (changes requested):
Good start so far 👍🏻 

Please also implement:
- the `createEvent` method
- comments explaining the methods `/** ... */`
- tests

Feel free to take the GoogleAnalytics adapter as reference => https://github.com/utopia-php/analytics/blob/main/src/Analytics/GoogleAnalytics.php

Pretty-19 commented:
-[x] Fixed comments :
        created the `createEvent` method
        added comments explaining the methods /** ... */
        Added Tests
 -[x] Submitted  pull request

Please have a check if the changes made are as requested. Thanks

christyjacob4 commented on tests/Analytics/AnalyticsTest.php:
```diff
@@ -31,6 +31,15 @@ public function testGoogleAnalytics()
         $this->assertTrue($this->ga->createPageView("appwrite.io", "/docs/installation"));
         $this->assertTrue($this->ga->createEvent("testEvent", "testEvent"));
 
+        $this->ga->disable();
+        $this->assertFalse($this->ga->createPageView("appwrite.io", "/docs/installation"));
+        $this->assertFalse($this->ga->createEvent("testEvent", "testEvent"));
+    }
+    public function testPlausible()
+    {
+        $this->assertTrue($this->ga->createPageView("appwrite.io", "/docs/installation"));
+        $this->assertTrue($this->ga->createEvent("testEvent", "testEvent"));
+
```
Look's like you're still using the Google Analytics adapter in your tests. You need to write these tests after creating a new instance of the Plausible adapter.

christyjacob4 commented on src/Analytics/Plausible.php:
```diff
@@ -0,0 +1,107 @@
+ * @license The MIT License (MIT) <http://www.opensource.org/licenses/mit-license.php>
+ */
+
+namespace Utopia\Analytics;
```
Please run this file through the [php-cs-fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer) to ensure consistency in indentation and formatting.

christyjacob4 reviewed (changes requested):
Please address the comments

Pretty-19 commented:
@christyjacob4  will do

christyjacob4 commented:
@Pretty-19  please let us know if you're still working on this 🙂

Pretty-19 commented:
@christyjacob4  yes working on this will push the changes

Pretty-19 commented:
@christyjacob4  can you have a check on the changes made

christyjacob4 commented:
@Pretty-19 thank you so much for the contribution!! 🥳
